src/k8s/pkg/component/network.go

```diff
@@ -98,7 +98,11 @@ func EnableNetworkComponent(ctx context.Context, s snap.Snap, podCIDR string) er
 			return fmt.Errorf("failed to get mount propagation for %s: %w", p, err)
 		}
 		if p == "private" {
-			if s.OnLXD(ctx) {
+			onLXD, err := s.OnLXD(ctx)
+			if err != nil {
+				return fmt.Errorf("failed to check if on lxd: %w", err)
```

Comment:
let's silently eat up the error here, or just log something. the user does not care that we couldn't check for LXD, they were trying to enable network